performance/plugins/extrinsicTransfer.ts

```diff
@@ -4,100 +4,53 @@ import { Mangata } from "@mangata-finance/sdk";
 import { TestParams } from "../testParams";
 import { KeyringPair } from "@polkadot/keyring/types";
 import { MGA_ASSET_ID } from "../../utils/Constants";
-import { preGenerateTransactions, runTransactions } from "./testRunner";
+import { runTransactions } from "./testRunner";
 import { performanceTestItem } from "./performanceTestItem";
-import { Commands } from "../testFactory";
 
 export class ExtrinsicTransfer extends performanceTestItem {
-  async arrange(numberOfThreads: number, nodes: string[]): Promise<boolean> {
-    await super.arrange(numberOfThreads, nodes);
-    await this.mintTokensToUsers(numberOfThreads, nodes, [MGA_ASSET_ID]);
+  async arrange(testParams: TestParams): Promise<boolean> {
+    await super.arrange(testParams);
+    await this.mintTokensToUsers(testParams.threads, testParams.nodes, [
+      MGA_ASSET_ID,
+    ]);
     return true;
   }
+
   async act(testParams: TestParams): Promise<boolean> {
-    let keepCalling = true;
-    setTimeout(function () {
-      keepCalling = false;
-      //duration in minutes, transform to millisecs.
-    }, testParams.duration * 60 * 1000);
-    while (keepCalling) {
-      let preSetupThreads;
-      if (testParams.command === Commands.ExtrinsicTransfer) {
-        preSetupThreads = await preGenerateTransactions(
-          testParams,
-          this.mgaNodeandUsers,
-          createAndSignTransfer
-        );
-      } else {
-        preSetupThreads = await preGenerateTransactions(
-          testParams,
-          this.mgaNodeandUsers,
-          createAndSignTransferKA
-        );
-      }
-      await runTransactions(testParams, preSetupThreads);
-    }
+    await super.act(testParams);
+    await runTransactions(
+      this.mgaNodeandUsers,
+      testParams,
+      createAndSignTransfer
+    );
     console.info(`.... Done Sending Txs`);
     return true;
   }
 }
 
 async function createAndSignTransfer(
-  mgaNodeandUsers: Map<
-    number,
-    { mgaSdk: Mangata; users: { nonce: BN; keyPair: KeyringPair }[] }
-  >,
-  nodeThread: number,
-  userNo: number
+  mgaSdk: Mangata,
+  users: { nonce: BN; keyPair: KeyringPair }[],
+  threadId: number,
```

Comment:
based on thread id createAndSignTransfer (and other in different scenarios) needs to generate tx, src user is defined by threadId parameter